pytransform3d/trajectories.py

```diff
@@ -109,6 +109,79 @@ def concat_many_to_one(A2Bs, B2C):
     return np.einsum("ij,njk->nik", B2C, A2Bs)
 
 
+def concat_many_to_many(A2B, B2C):
+    """Concatenate multiple transformations A2B with transformation B2C.
+
+    We use the extrinsic convention, which means that B2C is left-multiplied
+    to A2Bs.
+
+    Parameters
+    ----------
+    A2B : array-like, shape (n_transforms, 4, 4)
+        Transforms from frame A to frame B
+
+    B2C : array-like, shape (n_transforms, 4, 4)
+        Transform from frame B to frame C
+
+    Returns
+    -------
+    A2Cs : array, shape (n_transforms, 4, 4)
+        Transforms from frame A to frame C
+
+    See Also
+    --------
+    concat_many_to_one :
+        Concatenate one transformation with multiple transformations.
+
+    pytransform3d.transformations.concat :
+        Concatenate two transformations.
+    """
+    return np.einsum("ijk,ikl->ijl", B2C, A2B)
+
+
+def concat_dynamic(A2B, B2C):
+    """Concatenate multiple transformations A2B with transformation B2C.
```

Comment:
To clarify what "dynamic" means:

```suggestion
def concat_dynamic(A2B, B2C):
    """Concatenate multiple transformations A2B with transformation B2C.
    This function accepts different numbers or shapes of the input transformations, e.g. 4x4 or Nx4x4 shape.
```